File: Study/Intent/SubActivity.java
package com.example.intentex;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

public class SubActivity extends AppCompatActivity {

    private TextView tv_sub;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sub);

        tv_sub = findViewById(R.id.tv_sub);

        Intent intent = getIntent(); //SubActivity실행시 또 다른 인텐트 변수로 넘어온 인텐트를 받아옴.
        String str = intent.getStringExtra("str"); //인텐트에는 부가정보(str)이 담겨있으므로 이를 받아옴.

        tv_sub.setText(str);

    }
}

File: Study(Kotlin)/Intent/StartActivityForResult/SubActivity.kt
package com.example.intentsample

import android.app.Activity
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText

class SubActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_sub)


        val editText:EditText = findViewById(R.id.editText)
        val extras = intent.extras ?: null
        if(intent.hasExtra(EXTRA_MESSAGE)){
            val message = extras?.getString(EXTRA_MESSAGE)
            editText.setText(message)
        }


        val button: Button = findViewById(R.id.button)
        button.setOnClickListener {
            val str = editText.text.toString()
            val data = Intent()
            data.putExtra(RETURN_MESSAGE, str)
            setResult(Activity.RESULT_OK, data) //결과를 bundle객체와 함께 전달.
            finish()
        }
    }
}
